packages/create-sitecore-jss/src/templates/nextjs/next.config.js

```diff
@@ -29,6 +29,20 @@ const nextConfig = {
   // Enable React Strict Mode
   reactStrictMode: true,
 
+  // enable image provider domains and device sizes
+  images: {
+    // We use a custom loader function in the NextImage component which is required to be passed as a prop to the Image(next/image) component.
+    // See: https://nextjs.org/docs/api-reference/next/image#loader
+    // the config here is more of a vanity configuration as it does not affect the functionality.
+    loader: 'custom',
+    //IMPORTANT: 'path' is required to set to an appropriate value for the next/image to work.
+    path: jssConfig.sitecoreApiHost,
+    // deviceSizes - If you know the expected device widths of your users, you can specify a list of device width breakpoints.
+    // These widths are used when the next/image component uses layout="responsive" or layout="fill" to ensure the correct image is served for user's device.
+    // It is used to generate the srcset attribute for the image, using two sizes 300 and 100px max widths, respecting aspect ratio.
+    deviceSizes: [100, 300],
```

Comment:
Just some suggestions for comments
```suggestion
    // We use a custom loader function in the NextImage component (passed to next/image).
    // See: https://nextjs.org/docs/api-reference/next/image#loader
    // The config here is more of a vanity configuration as it does not affect the functionality, but is recommended by Vercel.
    loader: 'custom',
    // IMPORTANT: 'path' is required as this drives our custom loader when media URLs are relative.
    path: jssConfig.sitecoreApiHost,
    // deviceSizes - If you know the expected device widths of your users, you can specify a list of device width breakpoints.
    // These widths are used when the next/image component uses layout="responsive" or layout="fill" to ensure the correct image is served for user's device.
    // It is used to generate the srcset attribute for the image, using two sizes 300 and 100px max widths, respecting aspect ratio.
    // IMPORTANT: These sizes should match your Sitecore server-side allowlist. See /sitecore/config/*.config (search for 'allowedMediaParams')
    deviceSizes: [100, 300],
```